arelle/plugin/validate/EDINET/rules/gfm.py

```diff
@@ -1256,6 +1256,44 @@ def rule_gfm_1_3_31(
         )
 
 
+@validation(
+    hook=ValidationHook.XBRL_FINALLY,
+    disclosureSystems=[DISCLOSURE_SYSTEM_EDINET],
+)
+def rule_gfm_1_4_6(
+        pluginData: PluginValidationDataExtension,
+        val: ValidateXbrl,
+        *args: Any,
+        **kwargs: Any,
+) -> Iterable[Validation]:
+    """
+    EDINET.EC5700W: [GFM 1.4.6] Please correct the value of the link:arcroleRef attribute to
+    one specified in the XBRL 2.1 specification or the EDINET taxonomy.
+
+    GFM 1.4.6: The text preceding a sharp sign "#" in an xlink:href attribute of link:arcroleRef must
+    be a standard taxonomy or in a file recognised in the disclosure system.
+    """
+
+    for modelDocument in val.modelXbrl.urlDocs.values():
+        if pluginData.isStandardTaxonomyUrl(modelDocument.uri, val.modelXbrl):
+            continue
+        rootElt = modelDocument.xmlRootElement
+        for elt in rootElt.iterdescendants():
+            if hasattr(elt, 'localName') and elt.localName == "arcroleRef":
```

Comment:
I think `root.iterfind('.//*[local-name()="arcroleRef"]')` will do this more efficiently.